### PR TITLE
Update Inplayer.js SDK to accept custom config

### PR DESCRIPTION
**Pull request by borkopetrovicc**
## OVERVIEW

Update Inplayer.js SDK to accept custom config
https://www.notion.so/Update-Inplayer-js-SDK-to-accept-custom-config-de0928449c784d44b800dc8f1bf07090

## WHERE SHOULD THE REVIEWER START?

src/factories/request.ts

## ANY NEW DEPENDENCIES ADDED?

_List any new dependencies added._

- [ ] Does it work in IE >= 11?
- [ ] _Does it work in other browsers?_

## CHECKLIST

_Be sure all items are_ ✅ _before submitting a PR for review._

- [ ] Verify you bumped the lib version according to [Semantic Versioning Standards](http://semver.org)
- [ ] Verify you updated the CHANGELOG
- [ ] Verify this branch is rebased with the latest master

### CODE REVIEW

File: src/factories/request.ts [hunk 1]
```diff
@@ -32,6 +33,8 @@ export default class Request {
     this.authenticatedInstance.interceptors.request.use(
       this.createAuthInterceptor,
     );
+
+    this.customAxiosConfig = {}; // Initialize with an empty object
   }
 
   setInstanceConfig = (configEnv: Env): void => {
```
Comment (darkoatanasovski) [inline src/factories/request.ts]: we only need one instance of Axios, so you will need to pass a custom config as a second parameter here.

Comment (borkopetrovicc) [reply src/factories/request.ts]: 20125dfc8315b789747e3a27465bdcd929cd8101

**Review by mirovladimitrovski — DISMISSED** [review]

File: src/factories/request.ts [hunk 2]
```diff
@@ -47,6 +51,26 @@ export default class Request {
     );
   };
 
+  // Step 1: Create a new method `setAxiosConfig`
+  setAxiosConfig = (customConfig: AxiosRequestConfig) => {
```
Comment (darkoatanasovski) [inline src/factories/request.ts]: this method is no longer needed with this approach.

Comment (borkopetrovicc) [reply src/factories/request.ts]: 5ed40c508aa410cc3eeac06c35c265dadfd4928c

File: src/models/Config.ts [hunk 3]
```diff
@@ -145,5 +146,5 @@ export interface Request {
     path: string,
     headers?: Record<string, Record<string, unknown> | string | boolean>
   ): any;
-  setInstanceConfig(configEnv: Env): void;
+  setInstanceConfig(configEnv: Env, axiosConfig?: AxiosRequestConfig,): void;
```
Comment (darkoatanasovski) [inline src/models/Config.ts]:
```suggestion
  setInstanceConfig(configEnv: Env, axiosConfig?: AxiosRequestConfig): void;
```

File: src/factories/request.ts [hunk 4]
```diff
@@ -20,6 +20,7 @@ export default class Request {
   config: ApiConfig;
   basicInstance: AxiosInstance;
   authenticatedInstance: AxiosInstance;
+  customAxiosConfig?: AxiosRequestConfig;
```
Comment (darkoatanasovski) [inline src/factories/request.ts]:
```suggestion
```

**Review by darkoatanasovski — DISMISSED** [review]
@borkopetrovicc just accept the suggestions and merge

**Review by darkoatanasovski — APPROVED** [review]

**Pull request merged**